packages/gui/src/components/html/Renderer.tsx

```diff
@@ -8,14 +9,19 @@ interface Props {
 export function HtmlRenderer({ value }: Props) {
   const { tagName, attributes = {}, style = {}, children = [] } = value
   const Tag: any = value.tagName || 'div'
+
   return (
-    <Tag {...attributes} sx={{ ...toCSSObject(style) }}>
-      {children.map((child, i) => {
-        if (typeof child === 'string') {
-          return child
-        }
-        return <HtmlRenderer key={i} value={child} />
-      })}
-    </Tag>
+    <>
+      <HTMLFontTags style={value.style}/>
+      <Tag {...attributes} sx={{ ...toCSSObject(style) }}>
+        {children.map((child, i) => {
+          if (typeof child === 'string') {
+            return child
+          }
+          return <HtmlRenderer key={i} value={child} />
+        })}
+      </Tag>
```

Comment:
Since the renderer is recursive, we're gonna end up potentially adding quite a few stylesheets. Not sure if we think there's a better way to manage this but at least this way keeps the fonts tags encapsulated to the renderer. Open to changes though

Reply:
I think we should keep this out of the recursive renderer and have a different recursive component to avoid the duplication in style tags. In this case we probably don't want encapsulation since this will be useful in our codegen, too. I.e., #132 

```js
<>
  <HTMLFontTags {...stuff} />
  <HTMLRenderer {...stuff} />
</>
```

```js
const HTMLFontTags = ({ styles }) => {
  const fonts = getFonts(styles) // Recursive similarly to the toCSSObject function
  const hrefs = getFontStylesheets(fonts)
  return fonts.map(/* ... */)
}
```

Reply:
Thats all fine, but at that stage we'd be iterating over the html tree rather than the styles as the styles object is set at htmlRenderer level. Not an issue, just something to note as it might still require some adaption for codegen.

Reply:
Right right. Guess my pseudo code was awry there. Would apply for HTML+CSS codegen, and then CSS specific codegen will get to use the `getFontStylesheets` from its own recursive "plucking" function.